File: backend/app/schemas/book.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class BookCreate(BaseModel):
    title: str
    subtitle: str | None = None
    isbn: str | None = None
    publication_year: int | None = None
    format: str | None = None
    notes: str | None = None
    location_id: uuid.UUID | None = None


class Book(BaseModel):
    id: uuid.UUID
    title: str
    subtitle: str | None
    isbn: str | None
    publication_year: int | None
    format: str | None
    notes: str | None
    location_id: uuid.UUID | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/book.ts
export interface Book {
  id: string;
  title: string;
  subtitle: string | null;
  isbn: string | null;
  publication_year: number | null;
  format: string | null;
  notes: string | null;
  location_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface BookCreate {
  title: string;
  subtitle?: string | null;
  isbn?: string | null;
  publication_year?: number | null;
  format?: string | null;
  notes?: string | null;
  location_id?: string | null;
  author_id?: string | null;
}

export interface BookUpdate {
  title: string;
  subtitle?: string | null;
  isbn?: string | null;
  publication_year?: number | null;
  format?: string | null;
  notes?: string | null;
  location_id?: string | null;
}

export interface AuthorSummary {
  id: string;
  display_name: string;
}

export interface LocationSummary {
  id: string;
  name: string;
  description: string | null;
}

export interface BookDetail extends Book {
  authors: AuthorSummary[];
  location: LocationSummary | null;
}
